src/core/apiState.ts

```diff
@@ -132,9 +132,9 @@ export type MutationSubState<D extends BaseEndpointDefinition<any, any, any>> =
   | (({
       status: QueryStatus.fulfilled;
     } & WithRequiredProp<BaseMutationSubState<D>, 'data' | 'fulfilledTimeStamp'>) & { error: undefined })
-  | ({
+  | (({
       status: QueryStatus.pending;
-    } & BaseMutationSubState<D>)
+    } & BaseMutationSubState<D>) & { data?: undefined })
```

Comment:
Forces the return type of `result.data` from `string | undefined` to `undefined` in the unionTypes test, which is what we want. `data` should never be a string there as a new request is `{ isLoading: true, data: undefined, error: undefined `